src/Auth/BaseAuthenticate.php

```diff
@@ -120,9 +120,24 @@ protected function _findUser(string $username, ?string $password = null)
         }
 
         $passwordField = $this->_config['fields']['password'];
+        $hashedPassword = $result->get($passwordField);
+
+        if ($hashedPassword === null) {
+            // Waste time hashing the password, to prevent
+            // timing side-channels. However, don't hash
+            // null passwords as authentication systems
+            // like digest auth don't use passwords
+            // and hashing *could* create a timing side-channel.
+            if ($password !== null) {
+                $hasher = $this->passwordHasher();
+                $hasher->hash($password);
+            }
+
+            return false;
+        }
```

Comment:
Not ideal, shall I create a new method to hash and return false?

Reply:
First `if` could be modified to `if ($result === null || $result->get($passwordField) === null)`

Reply:
Is this in line with our guidelines? wasn't sure about making a call inside an if condition.

Reply:
But on password-less authentication systems, there is no password, so `$hashedPassword` is always `null`.

Currently, PR is completely wrong.

Reply:
Then I am worried why all our tests pass.

Reply:
Add before `if (!$hasher->check(...))`:

```php
if ($hashedPassword === null) {
    $hasher->hash($password);
    return false;
}
```